### PR TITLE
 unit tests for DeviceCacheManagerImpl 

### PR DESCRIPTION
**Pull request by GDLMadushanka**
## Automation tests
 - Unit tests 
   > 97%

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+    Unit tests for deviceCacheManagerImpl
+ */
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java]: It should be "DeviceCacheManagerImpl"

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    @Test(description = "test updating and getting a device in Cache",
+            dependsOnMethods = {"testAddExistingDeviceToCache"})
+    public void testUpdateDeviceInCache() {
+        devices.get(0).setName(UPDATE_NAME);
+        deviceCacheManager.updateDeviceInCache(deviceIds.get(0), devices.get(0), MultitenantConstants.SUPER_TENANT_ID);
+        Device tempDevice = deviceCacheManager.getDeviceFromCache(deviceIds.get(0), MultitenantConstants
+                .SUPER_TENANT_ID);
+        Assert.assertEquals(tempDevice.getName(), UPDATE_NAME);
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java]: Shall we add  another parameter for specifying the reason  in test cases? Shall we add in other places as well

**Review by Megala21 — CHANGES_REQUESTED** [review]
Please make the relevant changes

**Review by Megala21 — APPROVED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        DeviceManagementConfig config = DeviceConfigurationManager.getInstance().getDeviceManagementConfig();
+        config.setDeviceCacheConfiguration(configuration);
+        initializeCarbonContext();
+    }
+
+    private void initializeCarbonContext() throws IOException {
+        if (System.getProperty("carbon.home") == null) {
+            File file = new File("src/test/resources");
+            if (file.exists()) {
+                System.setProperty("carbon.home", file.getAbsolutePath());
```
Comment (pamod) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java:73]: Duplicated method ?

Comment (GDLMadushanka) [reply components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java:73]: +1 fixed in 9ef622da4d64b16865314ac54c0af3d1e208cbcc

**Pull request merged**